src/components/edit-movie-form/edit-movie-form.tsx

```diff
@@ -1,105 +1,205 @@
+    items.find((movie) => movie.id === posterId),
+  );
+  const initialValues = { ...initialValue, ...movie };
+  const { fetchData: fetchEditMovie } = useApiRequest(
+    'put',
+    API_BASE,
+    editMovie,
+  );
 
-    setValues({
+  const onSubmit = (values) => {
+    const body = {
       ...values,
-      [target.name]: value,
-    });
+      runtime: parseInt(values.runtime),
+      id: parseInt(values.id),
+    };
+    fetchEditMovie(undefined, body);
+    hideEdit();
+  };
+
+  const {
+    handleSubmit,
+    handleChange,
+    values,
+    errors,
+    touched,
+    resetForm,
+    setFieldValue,
+  } = useFormik({
+    initialValues,
+    onSubmit,
+    validationSchema,
+  });
+
+  const handleOnSelect = (selected) => {
+    setFieldValue('genres', selected);
   };
 
-  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
+  const handleOnCalendar = (data) => {
+    const formattedDate = moment(data).format('YYYY-MM-DD');
 
-    console.log(JSON.stringify(values, null, 2));
+    setFieldValue('release_date', formattedDate);
   };
 
   return (
     <EditMovieFormWrapper>
       <EditMovieFormContainer>
-        <CloseButton onClick={() => hideEdit()} />
+        <CloseButton onClick={hideEdit} />
         <EditMovieFormTitle>Edit Movie</EditMovieFormTitle>
         <EditMovieFormInner onSubmit={handleSubmit}>
-          <Input
-            topic
-            label="Title"
-            name="title"
-            type="text"
-            placeholder="Moana"
-            onChange={handleOnChange}
-            value={values.title}
-          />
-          <ReleaseDatePicker />
-          <Input
-            movie
-            label="Movie url"
-            name="movie"
-            type="text"
-            placeholder="https://www.moana.com"
-            onChange={handleOnChange}
-            value={values.movie}
-          />
-          <Input
-            rating
-            label="Rating"
-            name="rating"
-            type="text"
-            placeholder="7.8"
-            onChange={handleOnChange}
-            value={values.rating}
-          />
-          <Select onChange={handleOnChange} value={values.genre} name="genre" />
-          <Input
-            runtime
-            label="Runtime"
-            name="runtime"
-            type="text"
-            placeholder="minutes"
-            onChange={handleOnChange}
-            value={values.runtime}
-          />
-          <Input
-            overview
-            label="Overview"
-            name="overview"
+          <div>
+            <Input
+              label="Movie id"
+              name="id"
+              type="text"
+              onChange={handleChange}
+              value={values.id}
+              autoComplete="off"
+              disabled
+            />
+            {
+              <EditMovieFormError>
+                {touched.id && errors.id ? errors.id : ''}
```

Comment:
You can notice a very similar pattern with errors. Maybe a good idea will be to extract it to function?